### ISSUE TITLE
Odd Font Display Issues

### ISSUE THREAD

**Issue by Flavor-name** [crosswire/xiphos#1178]
![shot-2024-10-28_17-44-21](https://github.com/user-attachments/assets/60dfe857-5159-4a2a-acac-5c25ae5a7172)

In the above image several of the issues can be see:

1. Hebrew Script is not displayed within the program.
2.  The Strong's Greek file contains foreign non-Greek characters.

( In the background, the terminal window is running GDB.)

Previously, I had been having problems with installation and closed a thread citing an OS reinstall as a fix.
Issues persist.
I believe, though I don't have evidence for, Hacktivism related to repositories for Arch Linux.
Could be a local issue, like MITM, or possibly source files corruption, or both.
My repo mirror list favors Ukraine, which I have commented out to prevent downloads from suspect repos.

Sometimes my fresh install would be corrupted, or my module selections would fail to install.
As an example: My installed webkit2gtk is 4.1, yet Xiphos is using a vulnerable libwebkit2gtk 4.0.so..37.
Why is Xiphos ignoring webkit2gtk-4.1 in favor of a vulnerable version?
Who is building Xiphos for Arch Linux? Does the issue begin here with the build or is it at suspect mirrors?

A few issues related to my problems which could be from the vulnerable webkit2gtk4.0:
An app may gain unauthorized access to Local Network.
A non-privileged user may be able to modify restricted network settings.
An app may gain unauthorized access to Bluetooth.
An attacker may be able to force a device to disconnect from a secure network.
A malicious app with root privileges may be able to modify the contents of system files.

I live in Israel and am not in favor of Ukraine's Neo Nazi political structure.
I find it strange when the Son's of Israel work with known Nazi's and Muslims, people who seek an end to their life.
But this is the weirdness when outside interests seek to create trouble within and without for Israel.
These political issues are affecting computer use, I just want to be free to do my bible research.

Annoyed by the failure of module installs, I finally ran gdb. While gdb was running, I installed modules and none of the previously failing modules failed.
Over 1 GB of symbols just for webkit2gtk...WTF!
This made me focus on repo issues. I discovered Arch Linux in generating its mirror list, favors certain mirrors over others.
How is Ukraine always the fastest mirror? Is it possible to restrict bandwidth to force malignant repos to be weighted to the top of the list? Why do local israeli mirrors always fail.
DDosing competing repos to get the malignant repo to download from the suspect file provider, is my guess.
Is it possible to only deliver problem files to specific locales while providing clean files to everyone else?
Because I use Wifi most of the time, is it possible to interfere with the signal? Beam Forming interference to a radius of a target?

Remove the negative structures within your heart, one structure at a time, and what remains is nothing but love.

When the attack can be carried out with vulnerable library files, how would an inexperienced individual be able to discover the source of what and where the problems reside? Can a suspect repo force a dependency downgrade to create an attack surface?

How do I verify all of the files in use by Xiphos? Strace? GDB?
How do I capture, log, the file info as pacman is downloading it?

Are there any plans with the Xiphos project to make an AppImage for Linux?
This would include dependencies you expect your software to contain and eliminate possible downgrade issues.

**Comment by karlkleinpaste** [comment]
Font problems are due to not having adequate fonts installed. I use FreeSerif almost everywhere, and both Hebrew and Greek display fine.
![screenshot3](https://github.com/user-attachments/assets/145007e0-5ec3-4e06-876b-7496566f6e8c)

As to repo problems, we have nothing to say about that. What comes out of our efforts in github is mostly source tarballs and a Windows binary. Per-platform builds and repo hosting of those builds are the responsibility of those platforms.

Under Linux, the current release is using libwebkit2gtk 4.0 because we haven't produced a release in too long. We are staggering toward release and the current state of the tree (if you cloned the repo and built out of it) uses 4.1. We hope to release 4.3.0 before Christmas.

There is nothing political in this development along any axis of consideration, that will remain the case forever, and it has no place being mentioned in any bug report.

**Comment by domcorbex** [comment]
If you don't mind to do some distro-hopping, I suggest you to try Fedora Linux, as it's one of the if not THE best for CrossWire related Bible software.

**Comment by Flavor-name** [comment]
Are you saying none of these installed fonts are usable?
`
DejaVuMathTeXGyre.ttf                OpenSans-LightItalic.ttf
DejaVuSans-BoldOblique.ttf           OpenSans-Light.ttf
DejaVuSans-Bold.ttf                  OpenSans-Regular.ttf
DejaVuSansCondensed-BoldOblique.ttf  OpenSans-SemiBoldItalic.ttf
DejaVuSansCondensed-Bold.ttf         OpenSans-SemiBold.ttf
DejaVuSansCondensed-Oblique.ttf      Roboto-BlackItalic.ttf
DejaVuSansCondensed.ttf              Roboto-Black.ttf
DejaVuSans-ExtraLight.ttf            Roboto-BoldItalic.ttf
DejaVuSansMono-BoldOblique.ttf       Roboto-Bold.ttf
DejaVuSansMono-Bold.ttf              RobotoCondensed-BoldItalic.ttf
DejaVuSansMono-Oblique.ttf           RobotoCondensed-Bold.ttf
DejaVuSansMono.ttf                   RobotoCondensed-Italic.ttf
DejaVuSans-Oblique.ttf               RobotoCondensed-LightItalic.ttf
DejaVuSans.ttf                       RobotoCondensed-Light.ttf
DejaVuSerif-BoldItalic.ttf           RobotoCondensed-Regular.ttf
DejaVuSerif-Bold.ttf                 Roboto-Italic.ttf
DejaVuSerifCondensed-BoldItalic.ttf  Roboto-LightItalic.ttf
DejaVuSerifCondensed-Bold.ttf        Roboto-Light.ttf
DejaVuSerifCondensed-Italic.ttf      Roboto-MediumItalic.ttf
DejaVuSerifCondensed.ttf             Roboto-Medium.ttf
DejaVuSerif-Italic.ttf               Roboto-Regular.ttf
DejaVuSerif.ttf                      Roboto-ThinItalic.ttf
Hack-BoldItalic.ttf                  Roboto-Thin.ttf
Hack-Bold.ttf                        VeraBd.ttf
Hack-Italic.ttf                      VeraBI.ttf
Hack-Regular.ttf                     VeraIt.ttf
OpenSans-BoldItalic.ttf              VeraMoBd.ttf
OpenSans-Bold.ttf                    VeraMoBI.ttf
OpenSansCondensed-Bold.ttf           VeraMoIt.ttf
OpenSansCondensed-LightItalic.ttf    VeraMono.ttf
OpenSansCondensed-Light.ttf          VeraSeBd.ttf
OpenSans-ExtraBoldItalic.ttf         VeraSe.ttf
OpenSans-ExtraBold.ttf               Vera.ttf
OpenSans-Italic.ttf
`

**Comment by karlkleinpaste** [comment]
Depends entirely on what font you use as a default, and what per-language fonts you've chosen for Hebrew and Greek in Settings → Fonts → Sizes and Faces → Font Preferences.

It won't go hunting fonts for you randomly. You have to tell it what you want. FreeSerif happens to have the near-widest coverage of any font set I've encountered. Linux Libertine fonts are also quite good. Opinion varies as to some being displayed too thin or too thick.

I have 1200+ font packages installed: An early step for me any time I install a new OS update (for me, of Fedora) is "dnf install *-fonts*.noarch" because it ensures I have every font available that I could possibly want or need.

**Comment by Flavor-name** [comment]
I have never had an issue in the past with fonts not displaying. This is an issue for the past 6 months or so.

`I think I remember now, reading something about Arch using Pango which switched from .ttf to .OTB.`
I have to look at Pango now and see how it affects font display on my Arch system. :sigh:

I have never had in the past Chinese characters inside the Greek Strong's Module.

I'm not interested in Fedora, 2 GB .iso + 6 GB of updates, no thanks. I don''t think Fedora will run well on 4 GB of Ram and 2 tabs of Firefox in a hostile environment.

**Comment by karlkleinpaste** [comment]
I think that, whatever else is going on here, you're looking at a platform-specific problem. Xiphos has no problem finding and using Hebrew and Greek fonts. I see no Chinese characters in any Strong's module (though you didn't give specific reference for where, or what specific Strong's module). Thus, this isn't a Xiphos bug to be handled here. If Arch has damaged something by moving to a different font handler... That's on Arch.

**Comment by Flavor-name** [comment]
Strong's Greek number 3588 ho-ho.

Really, 3588 is Dr. Strongs method of identifying unknown words.
When Dr. Strong created his reference that word was not in the dictionary.
3588 is an opportunity for research.

`extra/pango 1:1.54.0-1 [installed]
    A library for layout and rendering of text`

`$ fc-match
NotoSans-Regular.ttf: "Noto Sans" "Regular"`

**Comment by Flavor-name** [comment]
OK. In the Xiphos app, I found an Arch package contaning freeserif.
Only Free Serif Regular works specifically in preferences>fonts>faces>hbo , not the other versions of Free Serif.
Free Serif Regular is not legible for me.
I found Arimo Nerd Font Regular to be very legible for reading.

I have been using Arch for almost a year and never had any issues regarding fonts in Xiphos, everything worked. This is a recent change, within the past several months. 
Looking in the bbs.arch forum there seems to be fonts as squares as a result of Pango being updated from 1.43 to 1.44, which involved those using bitmap fonts.

There still are issues with application functioning, but that might be an issue with webkit2gtk needing to be updated.
When I decide it is important for me to update webkit2gtk on my own with Xiphos I'll review source code and make, make install procedures for Arch. Of course, self make apps don't receive updates from Arch, I will lose automation.

While trying to find a font to use, the preferences dialog would hang, refuse to re-enter hbo, would not close properly when clicking the [x] but would close properly when clicking [close]. I had to exit and re-enter Preferences every time I made a change. When I clicked the [x] to close the preferences, the [edit] menu would not open Preferences, I had to close the Xiphos application and reopen.
